### PR TITLE
Fix races in CQ execution

### PR DESCRIPTION
**Pull request by otoolep**
This change refactors the CQ code such that "should run" is checked within the goroutine that the CQ will run, if it does run. This avoids the race, at the cost a very-short lived goroutine if the CQ is not run.

### CODE REVIEW

**Comment by otoolep** [comment]
@pauldix @toddboom @corylanou

**Comment by otoolep** [comment]
These races were flagged here:

https://circle-artifacts.com/gh/influxdb/influxdb/2332/artifacts/1/tmp/circle-artifacts.8UTeJCy/test_logs_race.txt

File: server.go [hunk 1]
```diff
@@ -3955,6 +3953,18 @@ func (s *Server) runContinuousQuery(cq *ContinuousQuery) {
 	cq.mu.Lock()
 	defer cq.mu.Unlock()
 
+	// Get server CQ parameters under lock, to avoid races.
+	s.mu.RLock()
```
Comment (pauldix) [inline server.go]: do we really need to lock on these? They can't ever get updated on a running process. They're set when the server object is created and never set again.

Comment (otoolep) [inline server.go]: Perhaps not. This may be rather pedantic. This was a drive-by change -- the real race is around `lastRun` which was being written and read from different goroutines.

Comment (otoolep) [inline server.go]: I can pull this change. I agree that these settings never change.

**Comment by toddboom** [comment]
yeah, this seems sane to me, even if it only could ever fail during testing. +1
